scripts/with-ec2.sh

```diff
@@ -0,0 +1,38 @@
+#! /bin/bash
+
+# Runs a command on the host within the lifespan of an EC2 instance.
+# Uses AWS CLI v2
+
+get-instance-ip() {
+    aws ec2 describe-instances \
+        --instance-ids "$1" \
+        --query 'Reservations[0].Instances[0].PublicIpAddress' \
+        --output text
+}
+
+# This is necessary because a public IP might not be
+# available immediately after the instance starts.
+wait-for-instance-ip() {
+    local variant="$(get-instance-ip "$1")";
+
+    if [[ "$variant" =~ [0-9] ]]; then
+        echo "$variant";
+    else
+        sleep 3;
+        wait-for-instance-ip "$1"
+    fi
+}
+
+call-with-active-ec2() {
```

Comment:
Do these need to be called after some kind of login has happened through the AWS CLI?

Reply:
Yes. Most CLI interactions assume you've performed setup using `aws configure`.